pandas/core/dtypes/missing.py

```diff
@@ -231,7 +231,11 @@ def _isna_ndarraylike(obj):
 
 
 def _isna_ndarraylike_old(obj):
-    values = getattr(obj, "_values", obj)
+    if not isinstance(obj, np.ndarray):
+        values = obj.to_numpy()
+    else:
+        values = obj
```

Comment:
@simonjayhawkins How would we get mypy to catch bugs like this? The problem was evidently that values (Categorical) got passed to _isna_string_dtype which expects an ndarray according to annotations

Reply:
Right now anything in the np.ndarray space resolved to Any, so we don't get type checking against it. Most likely need stubs or annotations for numpy to have this catch going forward

Reply:
Thanks @WillAyd. Is having np.* resolve to Any a conscious decision until more types get clarified?

Reply:
This is just how mypy works when you set the `ignore_missing_imports` option (see setup.cfg); otherwise mypy would fail that there are no annotations found in the np.* space